File: android/src/main/java/com/example/u2f_plugin_java/model/Device.java
package com.example.u2f_plugin_java.model;

import androidx.annotation.NonNull;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.io.Serializable;

public class Device implements Serializable {
    private String deviceName;
    private String manufacturerName;
    private String productName;
    private String version;
    private String serialNumber;
    private int deviceId;
    private int vendorId;
    private int productId;
    private int deviceClass;
    private int deviceSubclass;
    private int deviceProtocol;
    private int configurationCount;

    public Device() {

    }

    public Device(String deviceName, String manufacturerName, String productName, String version,
                  String serialNumber, int deviceId, int vendorId, int productId, int deviceClass,
                  int deviceSubclass, int deviceProtocol, int configurationCount) {
        this.deviceName = deviceName;
        this.manufacturerName = manufacturerName;
        this.productName = productName;
        this.version = version;
        this.serialNumber = serialNumber;
        this.deviceId = deviceId;
        this.vendorId = vendorId;
        this.productId = productId;
        this.deviceClass = deviceClass;
        this.deviceSubclass = deviceSubclass;
        this.deviceProtocol = deviceProtocol;
        this.configurationCount = configurationCount;
    }

    public static String serialize(Device device) {
        Gson gson = new GsonBuilder().setPrettyPrinting().create();
        return gson.toJson(device);
    }

    @NonNull
    @Override
    public String toString() {
        return "Device{" +
                "deviceName='" + deviceName + '\'' +
                ", manufacturerName='" + manufacturerName + '\'' +
                ", productName='" + productName + '\'' +
                ", version='" + version + '\'' +
                ", serialNumber='" + serialNumber + '\'' +
                ", deviceId=" + deviceId +
                ", vendorId=" + vendorId +
                ", productId=" + productId +
                ", deviceClass=" + deviceClass +
                ", deviceSubclass=" + deviceSubclass +
                ", deviceProtocol=" + deviceProtocol +
                ", configurationCount=" + configurationCount +
                '}';
    }

    public void setDeviceName(String deviceName) {
        this.deviceName = deviceName;
    }

    public void setManufacturerName(String manufacturerName) {
        this.manufacturerName = manufacturerName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public void setSerialNumber(String serialNumber) {
        this.serialNumber = serialNumber;
    }

    public void setDeviceId(int deviceId) {
        this.deviceId = deviceId;
    }

    public void setVendorId(int vendorId) {
        this.vendorId = vendorId;
    }

    public void setProductId(int productId) {
        this.productId = productId;
    }

    public void setDeviceClass(int deviceClass) {
        this.deviceClass = deviceClass;
    }

    public void setDeviceSubclass(int deviceSubclass) {
        this.deviceSubclass = deviceSubclass;
    }

    public void setDeviceProtocol(int deviceProtocol) {
        this.deviceProtocol = deviceProtocol;
    }

    public void setConfigurationCount(int configurationCount) {
        this.configurationCount = configurationCount;
    }

    public String getDeviceName() {
        return deviceName;
    }

    public String getManufacturerName() {
        return manufacturerName;
    }

    public String getProductName() {
        return productName;
    }

    public String getVersion() {
        return version;
    }

    public String getSerialNumber() {
        return serialNumber;
    }

    public int getDeviceId() {
        return deviceId;
    }

    public int getVendorId() {
        return vendorId;
    }

    public int getProductId() {
        return productId;
    }

    public int getDeviceClass() {
        return deviceClass;
    }

    public int getDeviceSubclass() {
        return deviceSubclass;
    }

    public int getDeviceProtocol() {
        return deviceProtocol;
    }

    public int getConfigurationCount() {
        return configurationCount;
    }
}

File: lib/models/device.dart

class Device {

  String? deviceName;
  String? manufacturerName;
  String? productName;
  String? version;
  String? serialNumber;
  int deviceId;
  int vendorId;
  int productId;
  int deviceClass;
  int deviceSubclass;
  int deviceProtocol;
  int configurationCount;

  /*

  final String deviceName = '';
  final String manufacturerName = '';
  final String productName = '';
  final String version = '';
  final String serialNumber = '';
  final int deviceId = 0;
  final int vendorId = 0;
  final int productId = 0;
  final int deviceClass = 0;
  final int deviceSubclass = 0;
  final int deviceProtocol = 0;
  final int configurationCount = 0;

  String deviceName;
  String manufacturerName;
  String productName;
  String version;
  String serialNumber;
  int deviceId;
  int vendorId;
  int productId;
  int deviceClass;
  int deviceSubclass;
  int deviceProtocol;
  int configurationCount;
  */

  Device({
    this.deviceName = '',
    this.manufacturerName = '',
    this.productName = '',
    this.version = '',
    this.serialNumber = '',
    this.deviceId = 0,
    this.vendorId = 0,
    this.productId = 0,
    this.deviceClass = 0,
    this.deviceSubclass = 0,
    this.deviceProtocol = 0,
    this.configurationCount = 0,
  });

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      deviceName: json['deviceName'] as String?,
      manufacturerName: json['manufacturerName'] as String?,
      productName: json['productName'] as String?,
      version: json['version'] as String?,
      serialNumber: json['serialNumber'] as String?,
      deviceId: json['deviceId'] as int,
      vendorId: json['vendorId'] as int,
      productId: json['productId'] as int,
      deviceClass: json['deviceClass'] as int,
      deviceSubclass: json['deviceSubclass'] as int,
      deviceProtocol: json['deviceProtocol'] as int,
      configurationCount: json['configurationCount'] as int,
    );
  }

  @override
  String toString() {
    return 'Device{deviceName: $deviceName, manufacturerName: $manufacturerName, productName: $productName, version: $version, serialNumber: $serialNumber, deviceId: $deviceId, vendorId: $vendorId, productId: $productId, deviceClass: $deviceClass, deviceSubclass: $deviceSubclass, deviceProtocol: $deviceProtocol, configurationCount: $configurationCount}';
  }
}